go/libraries/doltcore/merge/merge.go

```diff
@@ -145,6 +145,14 @@ func (merger *Merger) MergeTable(ctx context.Context, tblName string, opts edito
 
 		// Deleted in root or in merge, either a conflict (if any changes in other root) or else a fast-forward
 		if ancHasTable && (!rootHasTable || !mergeHasTable) {
+			if isCherryPick && rootHasTable && !mergeHasTable {
+				// TODO : this is either drop table or rename table case
+				// We can delete only if the table in current HEAD and parent commit contents are exact the same (same schema and same data);
+				// otherwise, return ErrTableDeletedAndModified
+				// We need to track renaming of a table --> the renamed table could be added as new table
+				return nil, &MergeStats{Operation: TableModified}, errors.New(fmt.Sprintf("error: %s table was renamed or dropped", tblName))
```

Comment:
Maybe make this error message very similar to the no schema changes allowed message? 

```suggestion
				return nil, &MergeStats{Operation: TableModified}, errors.New(fmt.Sprintf("schema changes not supported: %s table was renamed or dropped in cherry-pick commit", tblName))
```